src/parser/Parser.ts

```diff
@@ -1235,10 +1230,40 @@ export class Parser {
         });
     }
 
-    private exitWhile(): ExitWhileStatement {
-        let keyword = this.advance();
-
-        return new ExitWhileStatement({ exitWhile: keyword });
+    private exitStatement(): ExitStatement {
+        const exitToken = this.advance();
+        if (exitToken.kind === TokenKind.ExitWhile) {
+            // `exitwhile` is allowed in code, and means `exit while`
+            // use an ExitStatement that is nicer to work with by breaking the `exit` and `while` tokens apart
+
+            const exitText = exitToken.text.substring(0, 4);
+            const whileText = exitToken.text.substring(4);
+            const originalRange = exitToken.location.range;
+            const originalStart = originalRange.start;
+
+            const exitRange = util.createRange(
+                originalStart.line,
+                originalStart.character,
+                originalStart.line,
+                originalStart.character + 4);
+            const whileRange = util.createRange(
+                originalStart.line,
+                originalStart.character + 4,
+                originalStart.line,
+                originalStart.character + exitToken.text.length);
+
+            return new ExitStatement({
```

Comment:
OH! One more request. Since we're creating new tokens, we should probably replace the `exitfor` token in the `.tokens` array with these two new tokens, that way external callers have the correct list of tokens after the parse is finished.